File: Problem.py
# Time complexity: O(k^2)
# Space complexity: O(k)
# Works on leetcode: yes
# Approach: The basic idea is to iteratively update the array from the end to the beginning where the end starts from
# iterating through the rowindices.
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        dp = [1]*(rowIndex+1)
        for i in range(2,rowIndex+1):
            for j in range(i-1,0,-1):
                dp[j] += dp[j-1]
        return dp
        

File: problem.java
public class problem {
    public int c(int[] nums, int target){
        if(nums == null || nums.length ==0) return 0;
        int res = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(nums[i])){
                map.put(nums[i], map.get(nums[i])+1);
            }
            else{
                map.put(nums[i], 0);
            }
        }
        for(int i=0;i<nums.length;i++){
            int a = nums[i] - target;
            if(map.containsKey(a) && map.get(a) < 2){
                res++;
            }
        }
        return res;
    }
}

//time complexity O(n)
//space complexity O(n)